### PR TITLE
WIP: Assertions in Drawing Classifier C++ Data Preparation

### PR DESCRIPTION
**Pull request by shantanuchhabra**
This is dependent on #1490. Please review after #1490 is merged in.

### CODE REVIEW

**Comment by hoytak** [comment]
@shantanuchhabra -- could you rebase and update this commit?

File: src/unity/toolkits/drawing_classifier/data_preparation.cpp [hunk 1]
```diff
@@ -260,54 +269,86 @@ flex_nd_vec paint_stroke(
...
             for (int dr = -pad; dr <= pad; dr++) {
                 for (int dc = -pad; dc <= pad; dc++) {
+                    DASSERT_GE(num_values_in_sum, 0);
+                    DASSERT_GE(row + dr, 0);
+                    DASSERT_LE(row + dr, dimension);
+                    DASSERT_GE(col + dc, 0);
+                    DASSERT_LE(col + dc, dimension);
+                    DASSERT_GE((row+dr) * dimension + (col+dc), 0);
+                    DASSERT_LT((row+dr) * dimension + (col+dc), bitmap.num_elem());
+                    std::vector<size_t> indices = {0,(size_t)(row+dr),(size_t)(col+dc)};
```
Comment (hoytak) [inline src/unity/toolkits/drawing_classifier/data_preparation.cpp]: Also need to put this section in an `#ifndef NDEBUG ... #endif`

File: src/unity/toolkits/drawing_classifier/data_preparation.cpp [hunk 2]
```diff
@@ -320,7 +361,7 @@ flex_image blur_bitmap(flex_nd_vec bitmap, int ksize) {
 }
 
 #ifdef __APPLE__
-flex_image rasterize_on_mac(flex_list simplified_drawing) {
+flex_image rasterize_on_mac(flex_list &simplified_drawing) {
```
Comment (hoytak) [inline src/unity/toolkits/drawing_classifier/data_preparation.cpp]: Again, const

**Comment by shantanuchhabra** [comment]
Will address all feedback and put up a separate PR.

**Pull request closed without merging**